app/models/spree/order_contents_decorator.rb

```diff
@@ -1,5 +1,26 @@
 module Spree
   OrderContents.class_eval do
+
+    def add_to_line_item(variant, quantity, options = {})
+      line_item = grab_line_item_by_variant(variant, false, options)
+
+      if line_item && part_variants_match?(line_item, variant, quantity, options)
+        line_item.quantity += quantity.to_i
+        line_item.currency = currency unless currency.nil?
```

Comment:
Also, where is currency defined in this class? or should this be `order.currency` as in line 11?